### DISCUSSION TITLE
Problem closing the last Tab

### DISCUSSION THREAD

**Discussion by hlmaldon** [tabwrangler/tabwrangler#417] [General]
When setting up the number of Auto-Close tabs only if I have more than zero (0), the last tab does not close the browser.

All tabs close but the last one does not. Any suggestions?

![image](https://github.com/tabwrangler/tabwrangler/assets/123761774/1315ec0b-a72b-4961-81ff-a5c01fe6e6cc)

**Comment by ssorallen** [comment]
Currently Tab Wrangler prevents the active tab from being closed. This matches the "Closes inactive tabs after" intention because the active tab, by definition, is never inactive.

When Tab Wrangler correctly tracks browser activity in https://github.com/tabwrangler/tabwrangler/issues/361 then the definition of "inactive tab" could be updated to include "the active tab when the browser is inactive and inactive tabs".

https://github.com/tabwrangler/tabwrangler/blob/master/app/background.ts#L20-L21

**Comment by sergiu46** [comment]
Any updates on this topic?
Can the last tab be closed?

**Comment by ssorallen** [comment]
@sergiu46 Tab Wrangler will never close the active tab in the last focused window (usually the window on top). This is to prevent someone from accidentally creating a configuration where the browser immediately closes windows without letting the user interact.

The active tab in windows *other than* the last focused one are able to be closed, which means non-focused windows can be closed as well.

This is how Tab Wrangler intends to function for the foreseeable future.
